includes/classes/EnqueueScript.php

```diff
@@ -0,0 +1,284 @@
+		}
+	}
+
+	/**
+	 * Flag to decide whether load script in footer.
+	 *
+	 * @since x.x.x
+	 */
+	public function load_in_footer(): EnqueueScript {
+		$this->load_script_in_footer = true;
+
+		return $this;
+	}
+
+	/**
+	 * Set script dependencies.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param array $script_dependencies Script dependencies.
+	 */
+	public function dependencies( array $script_dependencies ): EnqueueScript {
+		$this->script_dependencies = $script_dependencies;
+
+		return $this;
+	}
+
+	/**
+	 * Register script.
+	 *
+	 * @since x.x.x
+	 */
+	public function register(): EnqueueScript {
+		$script_url   = $this->plugin_dir_url . $this->relative_script_path;
+		$script_asset = $this->get_asset_file_data();
+
+		$this->version = $script_asset['version'];
+
+		wp_register_script(
+			$this->script_id,
+			$script_url,
+			$script_asset['dependencies'],
+			$script_asset['version'],
+			$this->load_script_in_footer
+		);
+
+		if ( $this->register_translations ) {
+			wp_set_script_translations(
+				$this->script_id,
+				$this->text_domain,
+				$this->plugin_dir_path . 'languages'
+			);
+		}
+
+		if ( $this->localize_script_param_data ) {
+			wp_localize_script(
+				$this->script_id,
+				$this->localize_script_param_name,
+				$this->localize_script_param_data
+			);
+		}
+
+		return $this;
+	}
+
+	/**
+	 * This function should be called before enqueue or register function.
+	 *
+	 * @since x.x.x
+	 */
+	public function register_translations(): EnqueueScript {
+		$this->register_translations = true;
+
+		return $this;
+	}
+
+	/**
+	 * This function should be called after enqueue or register function.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param string $js_variable_name JS variable name.
+	 * @param array  $data             Data to be localized.
+	 */
+	public function register_localize_data( string $js_variable_name, array $data ): EnqueueScript {
+		$this->localize_script_param_name = $js_variable_name;
+		$this->localize_script_param_data = $data;
+
+		return $this;
+	}
+
+	/**
+	 * Enqueue script.
+	 *
+	 * @since x.x.x
+	 */
+	public function enqueue(): EnqueueScript {
+		if ( ! wp_script_is( $this->script_id, 'registered' ) ) {
+			$this->register();
+		}
+		wp_enqueue_script( $this->script_id );
+
+		return $this;
+	}
+
+	/**
+	 * Should return script ID.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return string
+	 */
+	public function get_script_id(): string {
+		return $this->script_id;
+	}
+
+	/**
+	 * Get asset file data.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return array
+	 */
+	public function get_asset_file_data(): array {
+		$script_asset_path = trailingslashit( dirname( $this->absolute_script_path ) )
+			. basename( $this->absolute_script_path, '.js' )
+			. '.asset.php';
+
+		$script_asset = file_exists( $script_asset_path ) ?
+			require $script_asset_path :
+			[
+				'dependencies' => [],
+				'version'      => $this->version ?: filemtime( $this->absolute_script_path ), // phpcs:ignore
```

Comment:
```suggestion
				'version'      => ! empty( $this->version ) ? $this->version : DT_VERSION,
```

On multi-server instances, the file time can vary between instances so best to fall back to the plugin version to improve cache hits. It will also be easier to get through [code review for VIP clients](https://docs.wpvip.com/technical-references/vip-platform/file-concatenation-and-minification/#h-versioning-to-bust-the-cache).